puppy.py

```diff
@@ -28,4 +29,177 @@ def __init__(self, behavior_matches):
         # the puppy.
         self.touch_sensor = TouchSensor(Port.S1)
 
+    def walk_in_place(puppy, speed=100):
```

Comment:
this should be defined walk_in_place(self,speed). then, below, instead of puppy.right_leg_motor, it should be self.right_leg_motor. same with puppy.left_leg_motor --> 
\self.left_leg_motor